ops/pebble.py

```diff
@@ -1899,6 +1899,21 @@ def get_services(self, names: Optional[Iterable[str]] = None) -> List[ServiceInf
         resp = self._request('GET', '/v1/services', query)
         return [ServiceInfo.from_dict(info) for info in resp['result']]
 
+    if typing.TYPE_CHECKING:
```

Comment:
I think we should drop the `typing.TYPE_CHECKING` tests unless they are actually needed, for simplicity/consistency. I know we haven't been quite consistent in the past. Thoughts?

Reply:
I've dropped these - there doesn't seem to be any value, indeed. The (stdlib) does imply that it's only for something that would be very expensive to do, so you want to skip it unless you're actually type checking. So that adds to the argument of getting rid of them.

There are some I'd definitely change, e.g. in `ops.log`:

```python
import typing

if typing.TYPE_CHECKING:
    from types import TracebackType
    from typing import Type

    from ops.model import _ModelBackend
```

`Type` can just be `typing.Type` since it's already imported. `types` will already be in `sys,modules` so it's just a cheap dictionary lookup. It doesn't seem like avoiding importing `ops.model` unless type checking has any real value (and `_ModelBackend` is already private so doesn't expand the public `log` namespace).

`ops.main` is similar:

```python
if TYPE_CHECKING:
    from ops.charm import CharmBase
    from ops.framework import BoundEvent, EventSource
    from ops.model import Relation
```

`charm`, `framework`, and `model` are all imported already, so all this does is avoid adding those to the public namespace. We're telling everyone to use everything from `ops` anyway, plus `__all__` is a more proper way to manage that, so I see no value in the conditional.

The other cases seem much the same. Should I open a separate PR that drops them everywhere they don't seem needed?

Reply:
Excellent digging, thanks. Yep, opening a separate PR that drops them everywhere sounds good.